### PR TITLE
GH-33800: [Packaging] Drop support for Ubuntu 18.04

### LINKED ISSUE #33800: [Packaging] Drop support for Ubuntu 18.04
### Describe the enhancement requested

Ubuntu 18.04 will reach End of Standard Support on 2023-04: https://wiki.ubuntu.com/Releases

> Version | Code name | Docs | Release | End of Standard Support | End of Life
> -- | -- | -- | -- | -- | --
> Ubuntu 18.04.6 LTS | Bionic Beaver | Changes | September 17.2021 | April 2023 | April 2028

We'll release 12.0.0 on 2023-04 so 12.0.0 doesn't need Ubuntu 18.04 support. We can drop support for Ubuntu 18.04 support now because the maintenance branch for 11.0.0 is already created.

FYI: We can require CMake 3.16 or later after we drop support for Ubuntu 18.04 because Ubuntu 20.04 ships CMake 3.16 and EPEL for CentOS 7 ships CMake 3.17.

### Component(s)

Packaging

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

Ubuntu 18.04 will reach End of Standard Support on 2023-04: https://wiki.ubuntu.com/Releases

> Version | Code name | Docs | Release | End of Standard Support | End of Life
> -- | -- | -- | -- | -- | --
> Ubuntu 18.04.6 LTS | Bionic Beaver | Changes | September 17.2021 | April 2023 | April 2028

### What changes are included in this PR?

Remove Ubuntu 18.04 related things.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

Yes.

* Closes: #33800

### CODE REVIEW

**Comment by kou** [comment]
@github-actions crossbow submit -g nightly-tests -g nightly-packaging -g nightly-release

**Comment by kou** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by kou** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by kou** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by assignUser** [comment]
Please don't merge the changes to r-binary-packages yet as I am fairly sure we need that version due to compiler versions etc. and can't replace it with 20.04. But can't check `nixlibs.R` right now. Right @nealrichardson ?

**Comment by kou** [comment]
Do we need to support R 3.6?
Do we need to support Debian buster?

If we can drop support for them, we can replace `ubuntu-18.04` binaries with binaries built on Ubuntu 20.04.

**Comment by nealrichardson** [comment]
> Do we need to support R 3.6? 

Yes, but that shouldn't matter for the libarrow binaries we build.
 
> Do we need to support Debian buster?

The libarrow binaries we build are optional. As long as arrow compiles everywhere it should, it's ok if we don't provide binaries that work there, they're just an installation speedup.

> If we can drop support for them, we can replace `ubuntu-18.04` binaries with binaries built on Ubuntu 20.04.

#32292 and #33091 should let us switch to using binaries built on CentOS 7, IIUC, so this would be moot. We'll basically have two versions of libarrow binaries: one built against openssl 1 and the other against openssl 3.

I would prefer to leave the r-binary-packages jobs unmodified here and remove the ubuntu-18.04 from it by addressing those other issues.

**Comment by kou** [comment]
OK. Then I work on #32292 and #33091 before this.

**Comment by kou** [comment]
#34048 is for #32292 and #33091.

**Comment by kou** [comment]
@github-actions crossbow submit -g nightly-release -g nightly-tests -g nightly-packaging

File: ci/docker/conda.dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 ARG arch=amd64
-FROM ${arch}/ubuntu:18.04
+FROM ${arch}/ubuntu:22.04
```
Comment (assignUser) [inline ci/docker/conda.dockerfile:19]: I don't mind using 22 but is there a reason we skip 20?

File: dev/tasks/linux-packages/apache-arrow-apt-source/apt/debian-buster/Dockerfile
Comment (assignUser) [inline dev/tasks/linux-packages/apache-arrow-apt-source/apt/debian-buster/Dockerfile:1]: Buster has also reached EOL for a while right?

**Review by assignUser — APPROVED** [review]
LGTM :+1:

File: ci/docker/conda.dockerfile (hunk 1 above)
Comment (kou) [reply ci/docker/conda.dockerfile:19]: No.
It's just no-reason to use Ubuntu 20.04. :-)

File: dev/tasks/linux-packages/apache-arrow-apt-source/apt/debian-buster/Dockerfile
Comment (kou) [reply dev/tasks/linux-packages/apache-arrow-apt-source/apt/debian-buster/Dockerfile:1]: Yes.
I should have removed this in #13470.

**Comment by kou** [comment]
@github-actions crossbow submit test-conda-python-3.10

**Pull request merged**